test/openssl/test_digest.rb

```diff
@@ -88,7 +88,7 @@ def test_sha2
   end
 
   def test_sha512_truncate
-    pend "SHA512_224 is not implemented" unless digest_available?('SHA512-224')
+    pend "SHA512_224 is not implemented" unless digest_available?('sha512-224')
```

Comment:
Why do you think that the `SHA512` is better than `sha512`? 

By checking the current situation about this topic with the following commands in this repository, I see the mixture of the usages of "SHA" and "sha". I don't feel that the changes in the `test/openssl/test_digest.rb` is related to this PR's subject. If we want to align all the parts with the name with upper or lower case in this repository, we can do it on another PR.

```
$ grep -r '['\''"]SHA' test/ | head
test/openssl/test_pkey_ec.rb:    signature = p256.sign("SHA256", data)
test/openssl/test_pkey_ec.rb:    assert_equal true, p256.verify("SHA256", signature, data)
test/openssl/test_pkey_ec.rb:    assert_equal true, p256.verify("SHA256", signature0, data)
test/openssl/test_pkey_ec.rb:    assert_equal false, p256.verify("SHA256", signature1, data)
test/openssl/test_ns_spki.rb:    spki.sign(key1, OpenSSL::Digest.new('SHA256'))
test/openssl/test_x509req.rb:    req = issue_csr(0, @dn, @rsa1024, OpenSSL::Digest.new('SHA256'))
test/openssl/test_x509req.rb:    req = issue_csr(0, @dn, @dsa512, OpenSSL::Digest.new('SHA256'))
test/openssl/test_x509req.rb:    req = issue_csr(0, @dn, @rsa1024, OpenSSL::Digest.new('SHA256'))
test/openssl/test_x509req.rb:    req = issue_csr(1, @dn, @rsa1024, OpenSSL::Digest.new('SHA256'))
test/openssl/test_x509req.rb:    req = issue_csr(0, @dn, @rsa1024, OpenSSL::Digest.new('SHA256'))
```

```
$ grep -r '['\''"]sha' test/ | head
test/openssl/test_pkcs7.rb:    psi = OpenSSL::PKCS7::SignerInfo.new(@ee1_cert, @rsa1024, "sha256")
test/openssl/test_x509store.rb:    ca1_crl1 = issue_crl(revoke_info, 1, now, now+1800, [], ca1_cert, ca1_key, "sha256")
test/openssl/test_x509store.rb:    ca1_crl2 = issue_crl(revoke_info, 2, now, now+1800, [], ca1_cert, ca1_key, "sha256")
test/openssl/test_x509store.rb:    ca2_crl1 = issue_crl(revoke_info, 1, now, now+1800, [], ca2_cert, ca2_key, "sha256")
test/openssl/test_x509store.rb:    ca2_crl2 = issue_crl(revoke_info, 2, now-100, now-1, [], ca2_cert, ca2_key, "sha256")
test/openssl/test_x509store.rb:                     ca1_cert, ca1_key, "sha256")
test/openssl/test_x509store.rb:                     ca1_cert, ca1_key, "sha256")
test/openssl/test_x509req.rb:    req1 = issue_csr(0, @dn, @rsa1024, "sha256")
test/openssl/test_x509req.rb:    req2 = issue_csr(0, @dn, @rsa1024, "sha256")
test/openssl/test_x509req.rb:    req3 = issue_csr(0, @dn, @rsa1024, "sha512")
```

Reply:
It's not one being better than the other, they're both valid:
```rb
oid = OpenSSL::ASN1::ObjectId.new("SHA512-224")
=> #<OpenSSL::ASN1::ObjectId:0x0000000128113c98 @indefinite_length=false, @tag=6, @tag_class=:UNIVERSAL, @tagging=nil, @value="SHA512-224">
oid.short_name
=> "SHA512-224"
oid.long_name
=> "sha512-224"
```
I expanded the rdoc for both `Cipher#name` and `Digest#name` to be more explicit about this - they both behave the same (`Cipher.ciphers` and `Digest.digests` too).

Reply:
> It's not one being better than the other, they're both valid:

OK. I assumed that. I am talking about a consistent style in this testing file or files including other testing files. And you see there are only the cases with upper case. I also want to see there are only essentially necessary changes in this PR.

```
$ grep $'[\'"]SHA' test/openssl/test_digest.rb 
    check_digest(OpenSSL::ASN1::ObjectId.new("SHA1"))
    assert_equal(sha224_a, OpenSSL::Digest.hexdigest('SHA224', "a"))
    assert_equal(sha256_a, OpenSSL::Digest.hexdigest('SHA256', "a"))
    assert_equal(sha384_a, OpenSSL::Digest.hexdigest('SHA384', "a"))
    assert_equal(sha512_a, OpenSSL::Digest.hexdigest('SHA512', "a"))
    assert_equal(sha224_a, encode16(OpenSSL::Digest.digest('SHA224', "a")))
    assert_equal(sha256_a, encode16(OpenSSL::Digest.digest('SHA256', "a")))
    assert_equal(sha384_a, encode16(OpenSSL::Digest.digest('SHA384', "a")))
    assert_equal(sha512_a, encode16(OpenSSL::Digest.digest('SHA512', "a")))
    pend "SHA512_224 is not implemented" unless digest_available?('SHA512-224')
    assert_equal(sha512_224_a, OpenSSL::Digest.hexdigest('SHA512-224', "a"))
    assert_equal(sha512_256_a, OpenSSL::Digest.hexdigest('SHA512-256', "a"))
    assert_equal(sha512_224_a, encode16(OpenSSL::Digest.digest('SHA512-224', "a")))
    assert_equal(sha512_256_a, encode16(OpenSSL::Digest.digest('SHA512-256', "a")))
    pend "SHA3 is not implemented" unless digest_available?('SHA3-224')
    assert_equal(OpenSSL::Digest.hexdigest('SHA3-224', ""), s224)
    assert_equal(OpenSSL::Digest.hexdigest('SHA3-256', ""), s256)
    assert_equal(OpenSSL::Digest.hexdigest('SHA3-384', ""), s384)
    assert_equal(OpenSSL::Digest.hexdigest('SHA3-512', ""), s512)
    check_digest(OpenSSL::ASN1::ObjectId.new("SHA224"))
    check_digest(OpenSSL::ASN1::ObjectId.new("SHA256"))
    check_digest(OpenSSL::ASN1::ObjectId.new("SHA384"))
    check_digest(OpenSSL::ASN1::ObjectId.new("SHA512"))

$ grep $'[\'"]sha' test/openssl/test_digest.rb

$ grep $'[\'"]MD5' test/openssl/test_digest.rb 
    @d1 = OpenSSL::Digest.new("MD5")
    assert_equal(bin, OpenSSL::Digest.digest('MD5', data))
    assert_equal(hex, OpenSSL::Digest.hexdigest('MD5', data))
    assert_equal("MD5", @d1.name, "name")
    assert_equal("MD5", @d2.name, "name")
    check_digest(OpenSSL::ASN1::ObjectId.new("MD5"))
    assert_equal OpenSSL::Digest::MD5, OpenSSL::Digest("MD5")

$ grep $'[\'"]md5' test/openssl/test_digest.rb
```

```suggestion
    pend "SHA512_224 is not implemented" unless digest_available?('SHA512-224')
```